app/src/main/java/com/vero/tasky/auth/presentation/login/LoginViewModel.kt

```diff
@@ -54,39 +55,34 @@ class LoginViewModel @Inject constructor(
     }
 
     private fun logIn() {
-        if (state.isEmailValid) {
-            val password = state.password
-            when (loginUseCases.validatePasswordUseCase(password)) {
-                PasswordValidationResult.SUCCESS -> {
-                    updateState(state.copy(isLoading = true))
-                    viewModelScope.launch {
-                        loginUseCases.loginUseCase(
-                            email = state.emailAddress,
-                            password = password
-                        ).onSuccess { user ->
-                            userPreferences.saveUser(user)
-                            channel.send(UiLoginEvent.OnLogIn)
-                        }.onFailure {
-                            updateState(state.copy(isLoading = false))
-                            showError(R.string.network_error_on_login)
-                        }
-                    }
-                }
-                PasswordValidationResult.TOO_SHORT -> {
-                    showError(R.string.password_is_too_short)
-                }
-                PasswordValidationResult.NO_UPPERCASE -> {
-                    showError(R.string.password_is_not_secure_uppercase)
-                }
-                PasswordValidationResult.NO_DIGIT -> {
-                    showError(R.string.password_is_not_secure_digit)
-                }
-                PasswordValidationResult.NO_LOWERCASE -> {
-                    showError(R.string.password_is_not_secure_lowercase)
-                }
+        EmailErrorHandler.validateEmail(
+            isEmailValid = state.isEmailValid,
+            onEmailIsValid = ::onEmailIsValid,
+            showError = ::showError
+        )
+    }
+
+    private fun onEmailIsValid() {
+        PasswordErrorHandler.onPasswordValid(
```

Comment:
Maybe `PasswordErrorHandler` as a name is a bit confusing, since it has the job here to also handle correct passwords. What about `PasswordValidator` instead?